gdrive

```diff
@@ -91,22 +93,37 @@ def upload(service, filepath):
         created_file = service.files().create(body=file_metadata,
                                               media_body=media,
                                               fields='id')
+        upload_loop(created_file)
+    except Exception as err:
+        print('An error occurred while uploading the file.')
+        print(err)
+
+def upload_loop(created_file):
+    try:
+        start_time = time.time()
+        last_time = time.time()
+        last_size = 0.0
         response = None
         print("Uploading 0%", end='\r')
         while response is None:
             status, response = created_file.next_chunk()
             if status:
-                print("Uploading %d%% - %s/%s" % interpret_progress(status), end='\r')
-
+                threading.Thread(target=interpret_progress,
```

Comment:
Are you starting a new thread for each iteration of the loop?! Sounds terrible

Reply:
Fixed by making sure there is only one logging thread going on.
Concurrent logging threads would have no benefits while skipping a log has no downsides.